Description: Define _DEFAULT_SOURCE in addition to _BSD_SOURCE.
 Also support kFreeBSD and the Hurd in this check.
Forwarded: not-yet
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2016-04-06

--- a/compiler.m4
+++ b/compiler.m4
@@ -25,9 +25,9 @@
 	CPPFLAGS="${CPPFLAGS}${CPPFLAGS:+ }-D__EXTENSIONS__ -DBSD_COMP"
 	;;
 
-    *-*-linux-*)
+    *-*-linux-*|*-gnu)
 	CPPFLAGS="${CPPFLAGS}${CPPFLAGS:+ }-D_XOPEN_SOURCE=600 -D_XOPEN_SOURCE_EXTENDED"
-	CPPFLAGS="${CPPFLAGS}${CPPFLAGS:+ }-D_BSD_SOURCE"
+	CPPFLAGS="${CPPFLAGS}${CPPFLAGS:+ }-D_BSD_SOURCE -D_DEFAULT_SOURCE"
 	;;
 
     *-*-aix*)
